gen/modules/general.h

```diff
@@ -279,3 +281,33 @@ struct user_desc {
 #define BLKSSZGET  0x1268
 #define BLKPBSZGET 0x127B
 #endif
+
+
+/* -- based on  sys/socket.h -- */
+
+struct msghdr {
+    void         *msg_name;
+    socklen_t     msg_namelen;
+    struct iovec *msg_iov; 
+    size_t        msg_iovlen;
+    void         *msg_control;
+    size_t        msg_controllen;
+    int           msg_flags;
+};
+
+struct cmsghdr {
+    __kernel_size_t cmsg_len;
+    int             cmsg_level;
+    int             cmsg_type;
+};
+
+
+#define	SCM_RIGHTS	0x01 /* rw: access rights (array of int) */
+#define SCM_CREDENTIALS 0x02 /* rw: struct ucred		*/
+#define SCM_SECURITY	0x03 /* rw: security label		*/
```

Comment:
These "rw:" comments appear to be from upstream, so please omit them.

Reply:
done